url888b6fd9b4_jmacglashan_commandsToTasks_tgz-pJ8-commands_tests_Model3TestsJ8/src/commands/data/TrainingElementParser.java

```diff
@@ -0,0 +1,103 @@
+package commands.data;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FilenameFilter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import burlap.oomdp.auxiliary.StateParser;
+import burlap.oomdp.core.Domain;
+
+public class TrainingElementParser {
+
+	TrajectoryParser tp;
+	
+	
+	public TrainingElementParser(Domain d, StateParser sp){
+		tp = new TrajectoryParser(d, sp);
+	}
+	
+	
+	public TrainingElement getTrainingElementFromString(String sdata){
+		
+		int firstNewLineIndex = sdata.indexOf("\n");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 26 does not find the substring and returns -1, the call to substring() on line 27 will throw IndexOutOfBoundsException. Note that the method is publicly accessible to external callers who might not be aware of the constraint on string sdata.